webapp/src/pages/Settings.tsx

```diff
@@ -31,29 +62,540 @@ const Settings: React.FC = () => {
+                ),
+              },
+            ],
+            partialConfig.pipelines?.[pipelineIndex]
+              ? partialConfig.pipelines
+              : resultingConfig.pipelines,
+            "name"
+          ),
+        });
+      }}
+    />
+  );
+
+  const handleCustomMetricUpdate = (checked: boolean, metricName: string) => {
+    checked
+      ? setPartialConfig({
+          ...partialConfig,
+          ...{
+            metrics: _.merge({}, resultingConfig.metrics, {
+              [metricName]: {
+                class_name: "datasets.load_metric",
+                kwargs: {
+                  path: metricName.toLowerCase(),
+                },
+                additional_kwargs: ADDITIONAL_KWARGS_CUSTOM_METRICS.includes(
+                  metricName
+                )
+                  ? { average: "weighted" }
+                  : {},
+              },
+            }),
+          },
+        })
+      : setPartialConfig({
+          ...partialConfig,
+          ...{
+            metrics: _.omit(resultingConfig.metrics, metricName),
+          },
+        });
+  };
+
+  const getProjectConfigSection = () => (
+    <Box display="flex" flexDirection="column">
+      {displaySectionTitle("General")}
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent="flex-start"
+        marginLeft={2}
+      >
+        {displayReadonlyFields("name", resultingConfig.name)}
+        {displayReadonlyFields(
+          "rejection class",
+          resultingConfig.rejection_class
+        )}
+      </Box>
+      <Box display="flex" flexDirection="row" marginLeft={2}>
+        {displaySubSectionTitle("Columns")}
+        {resultingConfig.columns &&
+          Object.entries(resultingConfig.columns).map(
+            ([field, value], index) => (
+              <Box key={index} gap={(theme) => theme.spacing(1)}>
+                {displayReadonlyFields(field, value)}
+              </Box>
+            )
+          )}
+      </Box>
+      {divider}
+      {displaySectionTitle("Dataset")}
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent="flex-start"
+        marginLeft={2}
+      >
+        {displayReadonlyFields(
+          "class name",
+          resultingConfig.dataset?.class_name
+        )}
+        {displayReadonlyFields("remote", resultingConfig.dataset?.remote)}
+      </Box>
+      {resultingConfig.dataset?.kwargs && (
+        <Box display="flex" flexDirection="row" marginLeft={2} gap={5}>
+          {displaySubSectionTitle("Kwargs")}
+          {Object.entries(resultingConfig.dataset?.kwargs).map(
+            ([field, value], index) => (
+              <Box key={index} gap={(theme) => theme.spacing(1)}>
+                {displayReadonlyFields(field, String(value))}
+              </Box>
+            )
+          )}
+        </Box>
+      )}
+      {resultingConfig.dataset?.args &&
+        resultingConfig.dataset?.args.length > 0 && (
```

Comment:
Detail, but we could shorten that up by relying on the fact that `0` is falsy, just like `undefined`.
```suggestion
      {resultingConfig.dataset?.args?.length && (
```

Reply:
Oh, Yes. thanks for this suggestion :)

Reply:
So, I tried simplifying the line to check the empty array, But 0 is rendered for the following reason in JS [For reference](https://javascript.plainenglish.io/javascript-tip-why-0-is-rendered-when-using-the-logical-and-operator-with-array-length-aae24d05c9d7). So, I reverted the change here. Please comment if you find any other simplified way to write this.

Reply:
Ha, that's true! Because we are inside a JSX content. Good catch! My solution would be to cast it to a `Boolean`. This has the same result as this article's suggestion to add a double `!!` in front, but I find it cleaner. What do you think?
```suggestion
      {Boolean(resultingConfig.dataset?.args?.length) && (
```